File: cpp2/pro_12.cpp
#include <string>
#include <vector>

using namespace std;

int GCD(int a, int b){
    if(a==0)    return b;
    return GCD(b%a,a);
}

int LCM(int a, int b){
    return a*b/GCD(a,b);
}

vector<int> solution(int n, int m) {
    vector<int> answer;
    answer.push_back(GCD(n,m));
    answer.push_back(LCM(n,m));
    return answer;
}

File: javascript/pro_12.js
function solution(record) {
    var answer = [];
    let user = {}
    let inout = [];
    for(let str of record){
        let [a,b,c] = str.split(' ');
        if(a==='Enter'){
            inout.push([b, '님이 들어왔습니다.'])
        }
        if(a==='Leave'){
            inout.push([b, '님이 나갔습니다.'])
        }
        if(c)   user[b] = c;
    }
    for(let temp of inout){
        let [a,b] = temp;
        answer.push(user[a]+b);
    }
    return answer;
}